src/core/image_resizer.js

```diff
@@ -0,0 +1,326 @@
+/* Copyright 2023 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { FeatureTest, ImageKind, shadow } from "../shared/util.js";
+
+const MIN_IMAGE_DIM = 2048;
+const MAX_IMAGE_DIM = 65536;
```

Comment:
Another question that I just had is even trying such a large value makes sense in general?

Is there any browser that would support a `65536 x 65536` pixel image, and would that really be meaningful anyway? Hence I wonder if we could just use a slightly lower value, e.g. half the current one, since that might slightly speed up the binary searching in `_guessMax0` by not trying "ridiculousness" large canvas-sizes?

Reply:
Chrome is fine with large dimensions but the main thing the total area:
https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/platform/graphics/skia/skia_utils.h;l=61-69?q=kMaxCanvasArea&ss=chromium%2Fchromium%2Fsrc

`_guessMax` is really fast to guess the max dimensions because the final image is very small (height is 1) but it's a way more slower when trying to guess the max area.
Anyway for Firefox, we'll get the max are thanks to the prefs, we could maybe hard code the above value for Chrome.